fix Issue 16627 - non-static structs with postblit/dtor fields now nested

MartinNowak opened the pull request:
- Since dlang/dmd#5500 buildPostblit/Dtor/OpAssign are run before
  the struct size is finalized, thereby now making structs nested
  that previously were not.
- Not sure if the old behavior was an intended feature. If not
  we might want to deprecate it.

WalterBright commented:
This is rather hackish to use the storage class to transmit a flag. I can feel the technical debt :-) Perhaps use a FUNCFLAG bit instead?

MartinNowak commented:
> I can feel the technical debt :-)

I'd be in favor to deprecating these non-static, but unnested, structs, requiring people to always explicitly use `static struct`. Can you clarify the original intention behind this? Was it just a simple PODs don't need a context optimization?

MartinNowak committed:
fix Issue 16627 - non-static structs with postblit/dtor fields now nested

- Since dlang/dmd#5500 buildPostblit/Dtor/OpAssign are run before
  the struct size is finalized, thereby now making structs nested
  that previously were not.
- Not sure if the old behavior was an intended feature. If not
  we might want to deprecate it.

MartinNowak commented:
Found that I can reuse the `generated` flag in FuncDeclaration.
Still clarifying the specs for the current behavior would be appreciated, it wasn't inferable from the code.

WalterBright commented:
> Can you clarify the original intention behind this?

I don't recall it. But being in the spec, it is clearly intentional, and seems like a good idea :-)
